TEST_APPS/testcases/nfc/nfc_cli_helper.py

```diff
@@ -0,0 +1,72 @@
+"""
+Copyright (c) 2017, Arm Limited and affiliates.
+SPDX-License-Identifier: Apache-2.0
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+# ice-tea cli commands decorator class
+
+
+from nfc_messages import NfcErrors
+import logging
+import icetea_lib.tools.asserts as asserts
+
+class CliHelper():
+    """
+    Helper method, checks the nfc SDK error-code for you, makes writing a negative test much easier
+    Example:
+        if (target_is_eeprom):
+            nfc_command("dev1", "start", expected_retcode=-2, expected_nfc_error= NfcErrors.nfc_err_unsupported)
+        else:
+            nfc_command("dev1", "start")
+    """
+
+    def nfc_command(self, k, cmd,  # pylint: disable=invalid-name
+                    wait=True,
+                    timeout=10,
+                    expected_retcode=0,
+                    asynchronous=False,
+                    report_cmd_fail=True,
+                    expected_nfc_error=NfcErrors.nfc_ok):
+        response = self.command(k, cmd, wait, timeout, expected_retcode, asynchronous, report_cmd_fail)
+        asserts.assertEqual(int(response.parsed['lastnfcerror']), expected_nfc_error.value)
+        return response
+
+    @staticmethod
+    def command_is(string, command):
+        return string.split(' ')[0] == command
+
+    @staticmethod
+    def debug_nfc_data(key, value):
+        """
+        print useful data values for the host/user {{in between}} easy to spot brackets.
+        """
+        text = "{{%s=%s}}" % (key, value)
```

Comment:
someone might put braces in the value and this will go horribly wrong and in the future and be a pain to find

Reply:
True, thanks - I am going to change those prints , because I underestimated the security risk before . Will also be more user-friendly with the test for buffers being equal and add that length check in places it might make the test failure and intent clearer.